src/server/debug.rs

```diff
@@ -400,6 +400,19 @@ impl Debugger {
         let handle = box_try!(get_cf_handle(self.engines.kv_engine.as_ref(), CF_RAFT));
         let store_ids = HashSet::<u64>::from_iter(store_ids);
 
+        let count_voters = |peers: &[Peer]| peers.iter().filter(|p| !p.get_is_learner()).count();
+
+        // Try best to keep committed logs not be overwritten after
```

Comment:
You can't know precisely whether a log has been committed. So you should keep as many logs as possible.